src/components/forms/DataLocation.tsx

```diff
@@ -0,0 +1,84 @@
+import React, {useCallback} from 'react';
+import {FormField, FormFieldTypes, FormValidators} from './forms';
+import {cloneDeep} from 'lodash';
+import {set} from 'lodash';
+import {Link} from 'react-router-dom';
+import {Styles} from '../../libs/theme';
+import {asIdAndDisplayText, SelectEntry} from './SelectOptionInterface';
+import {CloudProviders} from './CloudProviders';
+import {DataLocations} from "./DataLocations";
+import {ResearchStage} from "./ResearchStage";
+
+export type DataLocationInfo = {
+    cloudProvider: null | SelectEntry,
+    dataLocation: null | SelectEntry,
+    locationUrl: null | string,
+    researchStage: null | SelectEntry,
+};
+
+export type DataLocationComponentProps = {
+    idx: number,
+    location: DataLocationInfo,
+    onChange: ({idx, location}: { idx: number, location: DataLocationInfo }) => void,
+    onDelete: (idx: number) => void,
+};
+
+export const DataLocation = (props: DataLocationComponentProps) => {
+    const {idx, location, onChange, onDelete} = props;
+
+    const onDataLocationChange = useCallback(({key, value}: { key: string, value: string | SelectEntry }) => {
+        const newDataLocation = cloneDeep(location);
+        set(newDataLocation, key, value)
+        onChange({idx: idx, location: newDataLocation});
+    }, [location, idx, onChange]);
+
+    const deleteButton = <Link
```

Comment:
Why bind this to a const?

Reply:
I wasn't sure where this was going to ultimately land in the component, but now that it's settled, I can remove the const and embed this inline.
I do find the code a bit easier to read at line 44 with this, but we can always revert the inline back to a variable if we don't like it.

Reply:
I'm satisfied with how this reads with the const removed.